src/go/k8s/pkg/console/configmap.go

```diff
@@ -165,6 +170,39 @@ func (cm *ConfigMap) genEnterprise() (e Enterprise) {
 	return e
 }
 
+func (cm *ConfigMap) genCloud() CloudConfig {
+	if cm.consoleobj.Spec.Cloud == nil ||
+		cm.consoleobj.Spec.Cloud.PrometheusEndpoint == nil {
+		return CloudConfig{}
+	}
+	prometheus := cm.consoleobj.Spec.Cloud.PrometheusEndpoint
+	cc := CloudConfig{
+		PrometheusEndpoint: PrometheusEndpointConfig{
+			Enabled: prometheus.Enabled,
+			BasicAuth: struct {
```

Comment:
Why anonymous struct here, can we use PrometheusEndpointConfig.BasicAuth?

Reply:
I just copied what Martin had to stay in sync. Not a huge fan as well :(